core/src/main/scala/org/bykn/bosatsu/Evaluation.scala

```diff
@@ -623,7 +623,7 @@ case class Evaluation[T](pm: PackageMap.Typed[T], externals: Externals) {
    * that names resolve
    */
   private[this] val eval: Ref => Scoped =
-    Memoize.function[Ref, Scoped] {
+    Memoize.memoizeDagHashed[Ref, Scoped] {
```

Comment:
this needs to be thread safe because it is on the object and can be seen by many threads.